test/src/qa/hpcp/test_hpcp_generatesignals.py

```diff
@@ -0,0 +1,108 @@
+import random
+import numpy as np
+from collections import defaultdict
+
+from scipy.signal import spectrogram
+import essentia
+from essentia.standard import *
+
+from test_hpcp_parameters import *
+
+
+def getSignalsList(tests):
+    """
+    getSignalsList(['testWindowWholeRange', 'testFrameSize']])
+    """
+    signals = {}
+    signalsKey = []
+    
+    for test in tests:
+        for signalNeeded in test2SignalsMapping[test]:
+            if not signalNeeded in signalsKey:
+                signals[signalNeeded]=getSignal(signalNeeded)
+                signalsKey.append(signalNeeded)
+    
+    return signals, signalsKey
+    
+def getSignal(signalNeeded):
+    if signalNeeded.startswith('Tones'):
+        return (getTone(signalNeeded))
+    elif signalNeeded.startswith('White Noise'):
+        return (getWhiteNoise())
+    elif signalNeeded.startswith('LPF'):
+        return (getLPF(signalNeeded))
+    elif signalNeeded.startswith('BPF'):
+        return (getBPF(signalNeeded))
+    else:
+        print ('Error in getSignal')
+        return ([])
+    
+
+def getTone(signalNeeded):
+
```

Comment:
add a short description in comments to each signal generation method, e.g., "Generate an ascending sequence of pitches", etc.